lang/cs/Org.Apache.REEF.DistributedR/TaskRunner.cs

```diff
@@ -63,18 +65,24 @@ public void Start()
             this.toDriverThread.Start();
         }
 
-        public void Stop(bool wait = false)
+        internal void Stop()
         {
             Logr.Log(Level.Info, "Shutting down");
 
-            if (!wait)
+            Submit(ShutdownMsg.Factory());
+            this.cancelToken.WaitHandle.WaitOne(3000);
```

Comment:
Yes. However currently, the bridge will only shutdown after receiving the shutdown message and the last job is complete. There should probably be a shutdown after last job and force shutdown now. I implemented something quick to get development moving and we  need to spend some serious time in design.